### PR TITLE
awu/GizmoToolDirection

### PR DESCRIPTION
**Pull request by asunawu**
For [AB#1389](https://dev.azure.com/euclideon/57c3bd2a-8f94-4578-b195-2e4fa9d8f295/_workitems/edit/1389).
[AB#1659](https://dev.azure.com/euclideon/57c3bd2a-8f94-4578-b195-2e4fa9d8f295/_workitems/edit/1659)

Fixed local space translate, the world space translate still needs to be improved.

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -2453,7 +2453,15 @@ void vcMain_RenderSceneWindow(vcState *pProgramState)
...
 
         //read direction axes again.
-        vcGIS_GetOrthonormalBasis(pProgramState->geozone, pProgramState->camera.position, &pProgramState->gizmo.direction[2], &pProgramState->gizmo.direction[1], &pProgramState->gizmo.direction[0]);
+        if (pProgramState->gizmo.operation == vcGO_Scale || pProgramState->gizmo.coordinateSystem == vcGCS_Local)
+        {
+          pProgramState->gizmo.direction[0] = udDouble3::create(1, 0, 0);
+          pProgramState->gizmo.direction[1] = udDouble3::create(0, 1, 0);
+          pProgramState->gizmo.direction[2] = udDouble3::create(0, 0, 1);
+        }
+        else
```
Comment (samsinsane) [inline src/main.cpp:2462]: The else should also have braces - the coding standard says something along the lines of: if the `if` has braces the `else` must have braces too.

File: src/vcCamera.cpp [hunk 2]
```diff
@@ -386,7 +386,22 @@ void vcCamera_HandleSceneInput(vcState *pProgramState, udDouble3 oscMove, udFloa
...
+    vcSceneItem *pItem = (vcSceneItem *)clickedItemRef.pItem->pUserData;
+    if (pItem != nullptr)
+    {
+      if (pProgramState->gizmo.operation == vcGO_Scale || pProgramState->gizmo.coordinateSystem == vcGCS_Local)
+      {
+        pProgramState->gizmo.direction[0] = udDouble3::create(1, 0, 0);
+        pProgramState->gizmo.direction[1] = udDouble3::create(0, 1, 0);
+        pProgramState->gizmo.direction[2] = udDouble3::create(0, 0, 1);
+      }
+      else
```
Comment (samsinsane) [inline src/vcCamera.cpp:401]: Braces here too.

File: src/main.cpp (hunk 1 above)
Comment (asunawu) [reply src/main.cpp:2462]: fixed.

File: src/vcCamera.cpp (hunk 2 above)
Comment (asunawu) [reply src/vcCamera.cpp:401]: fixed.

**Pull request merged**